Test failure `test_drop`

```
===================================
__________________________________ test_drop ___________________________________

dspath = '/tmp/datalad_temp_test_drop3madglh0'
records = '/tmp/datalad_temp_test_drop8s8mzg2j'

    @skip_if_on_windows  # see gh-61
    @with_tempfile
    @with_tempfile(mkdir=True)
    def test_drop(dspath=None, records=None):
        make_datarecord_zips('12345', records)
        ds = create(dspath, **ckwa)
        ds.ukb_init(
            '12345',
            ['20227_2_0', '25747_2_0', '25748_2_0', '25748_3_0'], **ckwa)
        ds.config.add('datalad.ukbiobank.keyfile', 'dummy', where='local')
        bin_dir = make_ukbfetch(ds, records)
    
        # baseline
        with patch.dict('os.environ', {'PATH': '{}:{}'.format(
                str(bin_dir),
                os.environ['PATH'])}):
            ds.ukb_update(merge=True, force=True, **ckwa)
        zips_in_ds = list(ds.pathobj.glob('**/*.zip'))
        neq_(zips_in_ds, [])
    
        # drop archives
        with patch.dict('os.environ', {'PATH': '{}:{}'.format(
                str(bin_dir),
                os.environ['PATH'])}):
            ds.ukb_update(merge=True, force=True, drop='archives', **ckwa)
        # no ZIPs can be found, also not in the annex
        eq_(list(ds.pathobj.glob('**/*.zip')), [])
        # we can get all we want (or rather still have it)
>       assert_status('notneeded', ds.get('.', **ckwa))

/opt/hostedtoolcache/Python/3.7.15/x64/lib/python3.7/site-packages/datalad_ukbiobank/tests/test_update.py:212: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

label = ['notneeded'], results = []

    def assert_status(label, results):
        """Verify that each status dict in the results has a given status label
    
        `label` can be a sequence, in which case status must be one of the items
        in this sequence.
        """
        label = ensure_list(label)
        results = ensure_result_list(results)
        if len(results) == 0:
            # If there are no results, an assertion about all results must fail.
>           raise AssertionError("No results retrieved")
E           AssertionError: No results retrieved


```

--- a/datalad_ukbiobank/tests/test_update.py
+++ b/datalad_ukbiobank/tests/test_update.py
@@ -71,7 +71,7 @@ def test_base(dspath=None, records=None, clonedir=None):
         '12345',
         ['20227_2_0', '25747_2_0', '25748_2_0', '25748_3_0'], **ckwa)
     # dummy key file, no needed to bypass tests
-    ds.config.add('datalad.ukbiobank.keyfile', 'dummy', where='local')
+    ds.config.add('datalad.ukbiobank.keyfile', 'dummy', scope='local')
 
     # fake ukbfetch
     bin_dir = make_ukbfetch(ds, records)
@@ -108,7 +108,7 @@ def test_base(dspath=None, records=None, clonedir=None):
         ds.ukb_update(merge=True, **ckwa)
 
     cloned = clone(source=ds.path, path=clonedir, **ckwa)
-    cloned.config.add('datalad.ukbiobank.keyfile', 'dummy', where='local')
+    cloned.config.add('datalad.ukbiobank.keyfile', 'dummy', scope='local')
     with patch.dict('os.environ', {'PATH': '{}:{}'.format(
             str(bin_dir),
             os.environ['PATH'])}):
@@ -139,7 +139,7 @@ def test_bids(dspath=None, records=None):
         ['20227_2_0', '25747_2_0', '25748_2_0', '25748_3_0'],
         bids=True, **ckwa)
     # dummy key file, no needed to bypass tests
-    ds.config.add('datalad.ukbiobank.keyfile', 'dummy', where='local')
+    ds.config.add('datalad.ukbiobank.keyfile', 'dummy', scope='local')
     bin_dir = make_ukbfetch(ds, records)
 
     # put fake ukbfetch in the path and run
@@ -190,7 +190,7 @@ def test_drop(dspath=None, records=None):
     ds.ukb_init(
         '12345',
         ['20227_2_0', '25747_2_0', '25748_2_0', '25748_3_0'], **ckwa)
-    ds.config.add('datalad.ukbiobank.keyfile', 'dummy', where='local')
+    ds.config.add('datalad.ukbiobank.keyfile', 'dummy', scope='local')
     bin_dir = make_ukbfetch(ds, records)
 
     # baseline

--- a/datalad_ukbiobank/tests/test_update.py
+++ b/datalad_ukbiobank/tests/test_update.py
@@ -208,16 +208,18 @@ def test_drop(dspath=None, records=None):
         ds.ukb_update(merge=True, force=True, drop='archives', **ckwa)
     # no ZIPs can be found, also not in the annex
     eq_(list(ds.pathobj.glob('**/*.zip')), [])
-    # we can get all we want (or rather still have it)
-    assert_status('notneeded', ds.get('.', **ckwa))
 
     # now drop extracted content instead
     with patch.dict('os.environ', {'PATH': '{}:{}'.format(
             str(bin_dir),
             os.environ['PATH'])}):
         ds.ukb_update(merge=True, force=True, drop='extracted', **ckwa)
+    test_extractfile = ds.pathobj / '20227_2_0' / 'fMRI' / 'rfMRI.nii.gz'
+    # test file is gone
+    assert not test_extractfile.exists()
     eq_(list(ds.pathobj.glob('**/*.zip')), zips_in_ds)
     # we can get all
     assert_status('ok', ds.get('.', **ckwa))
+    eq_(test_extractfile.read_text(), 'rfMRI.nii.gz')
     # a non-zip content file is still around
     eq_((ds.pathobj / '25747_2_0.adv').read_text(), '25747_2_0.adv')

--- a/datalad_ukbiobank/ukb2bids.py
+++ b/datalad_ukbiobank/ukb2bids.py
@@ -42,7 +42,6 @@ def restructure_ukb2bids(ds, subid, unrecognized_dir, base_path=None):
             annex=None,
             untracked='all',
             eval_subdataset_state='no',
-            report_filetype='raw',
             return_type='generator',
             result_renderer=None):
         path = Path(fp['path'])

--- a/datalad_ukbiobank/update.py
+++ b/datalad_ukbiobank/update.py
@@ -17,7 +17,7 @@ import shutil
 from datalad.interface.base import Interface
 from datalad.interface.utils import eval_results
 from datalad.interface.base import build_doc
-from datalad.interface.add_archive_content import AddArchiveContent
+from datalad.local.add_archive_content import AddArchiveContent
 from datalad.support.constraints import (
     EnsureChoice,
     EnsureStr,

--- a/datalad_ukbiobank/update.py
+++ b/datalad_ukbiobank/update.py
@@ -229,7 +229,7 @@ class Update(Interface):
                     AddArchiveContent.__call__(
                         props['key'],
                         key=True,
-                        annex=repo,
+                        dataset=ds,
                         # --use-current-dir due to
                         # https://github.com/datalad/datalad/issues/3995
                         use_current_dir=True,

--- a/tox.ini
+++ b/tox.ini
@@ -1,0 +1,11 @@
+[pytest]
+filterwarnings =
+    error::DeprecationWarning:^datalad
+    error:.*yield tests:pytest.PytestCollectionWarning
+    ignore:distutils Version classes are deprecated:DeprecationWarning
+    # comes from boto
+    ignore:the imp module is deprecated
+    # workaround for https://github.com/datalad/datalad/issues/6307
+    ignore:The distutils package is deprecated
+markers =
+    skip_if_on_windows

